lib/logstash/outputs/syslog.rb

```diff
@@ -125,17 +125,17 @@ class LogStash::Outputs::Syslog < LogStash::Outputs::Base
   # syslog message format: you can choose between rfc3164 or rfc5424
   config :rfc, :validate => ["rfc3164", "rfc5424"], :default => "rfc3164"
 
+  default :codec, nil
+
   def register
     @client_socket = nil
 
     if ssl?
       @ssl_context = setup_ssl
     end
-    
-    if @codec.instance_of? LogStash::Codecs::Plain
-      if @codec.config["format"].nil?
-        @codec = LogStash::Codecs::Plain.new({"format" => @message})
-      end
+
+    if @codec.nil?
```

Comment:
@tsaarni why default`nil` assigment and then the `nil`(as if nil is a special value placeholder constant) instead of using the same path suggested in https://github.com/elastic/logstash/pull/11401/files which was listed in the issue you reported (https://github.com/elastic/logstash/issues/11434) ?

Assigning the codec to `nil` to then change it `register` time is not a good practice.

Reply:
I used `nil` as a special value placeholder in attempt to avoid class identity comparison altogether.  Comparing class identity by their string representation seemed bit awkward to me, but I'm happy with any fix as long as it works! :) 

There already is a string-based alternative submitted here few years back but for some reason it was not merged https://github.com/logstash-plugins/logstash-output-syslog/pull/55. If the original submitter @jsvd is around maybe he could reconsider resurrecting the PR, or I can re-submit his fix here and add him as co-author.

Reply:
BTW: What I _really_ wanted to do was to set the `default :codec` to `LogStash::Codecs::Plain.new({ "format" => @message })` but I could not achieve that, because obviously I have no access to instance variable `@message` when defining default codec for the class. If that was possible, it would be possible to avoid instantiating codec in `register` like introduced here long time ago https://github.com/logstash-plugins/logstash-output-syslog/commit/2c73283fd25c093d35e7ace20d7a54685817d06f (change which introduced the class identity comparison).

Reply:
If you run tests on actual main, it fails for that check. To get the test back to green the string comparison works fine like in #65 which is why also this PR is green.

Reply:
@jsvd confirmed that the #55 was closed for some other reason, mainly because the Travis CI changes were already included.